datashare-app/src/main/java/org/icij/datashare/web/TaskResource.java

```diff
@@ -249,17 +248,39 @@ public Payload stopAllTasksPreflight() {
     @ApiResponse(responseCode = "200", description = "returns 200 and the created task", useReturnTypeSchema = true)
     @Post("/findNames/:pipeline")
     public List<TaskView<?>> extractNlp(@Parameter(name = "pipeline", description = "name of the NLP pipeline to use", in = ParameterIn.PATH) final String pipelineName, final OptionsWrapper<String> optionsWrapper, Context context) {
-        Properties mergedProps = propertiesProvider.createOverriddenWith(optionsWrapper.getOptions());
-        mergedProps.put(NLP_PIPELINE_OPT, pipelineName);
-        syncModels(parseBoolean(mergedProps.getProperty("syncModels", "true")));
+        Properties properties = applyProjectProperties(optionsWrapper);
+        properties.put(NLP_PIPELINE_OPT, pipelineName);
+        syncModels(parseBoolean(properties.getProperty("syncModels", "true")));
         List<TaskView<?>> tasks = new LinkedList<>();
-        if (parseBoolean(mergedProps.getProperty("resume", "true"))) {
-            tasks.add(taskManager.startTask(taskFactory.createEnqueueFromIndexTask((User) context.currentUser(), mergedProps)));
+        if (parseBoolean(properties.getProperty("resume", "true"))) {
+            tasks.add(taskManager.startTask(taskFactory.createEnqueueFromIndexTask((User) context.currentUser(), properties)));
         }
-        tasks.add(taskManager.startTask(taskFactory.createNlpTask((User) context.currentUser(), mergedProps)));
+        tasks.add(taskManager.startTask(taskFactory.createNlpTask((User) context.currentUser(), properties)));
         return tasks;
     }
 
+    public Properties applyProjectProperties(OptionsWrapper optionsWrapper) {
+        Properties properties = propertiesProvider.createOverriddenWith(optionsWrapper.getOptions());
+        return TaskResource.applyProjectTo(properties);
+    }
+
+    public static Properties applyProjectTo(Properties properties) {
+        Properties clone = (Properties) properties.clone();
+        clone.setProperty(QUEUE_NAME_OPTION, getQueueNameFor(properties));
+        clone.setProperty(MAP_NAME_OPTION, getReportMapNameFor(properties));
+        return clone;
+    }
+
+    public static String getQueueNameFor(Properties properties) {
+        String projectName = properties.getOrDefault("defaultProject","local-datashare").toString();
+        return "extract:queue:" + projectName;
```

Comment:
The return value is the whole queue name or is it just the prefix ?

Reply:
Just a prefix, hence why we still need to delete all candidates values: https://github.com/ICIJ/datashare/pull/1334/files#diff-eccbec11c2a85c47442becddc5cb5d77c93a4c44eaf29565d6fe51a00a19d6d3R237-R240